map_view/display-device-location-with-autopan-modes/src/main/java/com/esri/samples/display_device_location_with_autopan_modes/DisplayDeviceLocationWithAutopanModesSample.java

```diff
@@ -0,0 +1,200 @@
+
+      // create an ArcGISMap with the imagery basemap
+      ArcGISMap map = new ArcGISMap(Basemap.createImagery());
+
+      // create a view and add the ArcGISMap to it
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // create a combo box
+      ComboBox<String> comboBox = new ComboBox<>();
+      comboBox.setMaxWidth(Double.MAX_VALUE);
+      comboBox.setDisable(true);
+      // add the autopan modes to the combo box
+      comboBox.getItems().addAll("Recenter", "Navigation", "Compass", "Off");
+      comboBox.setValue("Recenter");
+
+      // add a label
+      Label autopanModeLabel = new Label("Choose an autopan mode:");
+      // add a checkbox that toggles the visibility of the location display
+      CheckBox checkBox = new CheckBox("Show device location");
+      checkBox.setDisable(true);
+
+      // access the json of the location points
+      String polylineData = IOUtils.toString(getClass().getResourceAsStream(
+        "/display_device_location_with_autopan_modes/polyline_data.json"), StandardCharsets.UTF_8);
+      // create a polyline from the location points
+      Polyline locations = (Polyline) Geometry.fromJson(polylineData, SpatialReferences.getWgs84());
+
+      // create a simulated location data source
+      SimulatedLocationDataSource simulatedLocationDataSource = new SimulatedLocationDataSource();
+      // set the location of the simulated location data source with simulation parameters to set a consistent velocity
+      simulatedLocationDataSource.setLocations(
+        locations, new SimulationParameters(Calendar.getInstance(), 5.0, 0.0, 0.0));
+
+      // configure the map view's location display to follow the simulated location data source
+      LocationDisplay locationDisplay = mapView.getLocationDisplay();
+      locationDisplay.setLocationDataSource(simulatedLocationDataSource);
+      locationDisplay.setAutoPanMode(LocationDisplay.AutoPanMode.RECENTER);
+      locationDisplay.setInitialZoomScale(1000);
+
+      // toggle the location display visibility on check
+      checkBox.setOnAction(event -> {
+        if (checkBox.isSelected()) {
+          // start the location display
+          locationDisplay.startAsync();
+
+        } else {
+          // turn off the location display
+          locationDisplay.stop();
+        }
+        // toggle the combo box interactions
+        comboBox.setDisable(!checkBox.isSelected());
+      });
+
+      // set the autopan mode of the location display based on the mode chosen from the combo box
+      comboBox.getSelectionModel().selectedItemProperty().addListener(e -> {
+
+        // set the scale that the map view will zoom to when the autopan mode is changed
+        locationDisplay.setInitialZoomScale(1000);
+
+        switch (comboBox.getSelectionModel().getSelectedItem()) {
+          case "Off":
+            locationDisplay.setAutoPanMode(LocationDisplay.AutoPanMode.OFF);
+            break;
+          case "Recenter":
```

Comment:
just to keep this naming in keeping with how the API documentation has it and the readme. 
```suggestion
          case "Re-center":
```